bpf/bpf_lxc.c

```diff
@@ -691,10 +693,13 @@ int handle_ingress(struct __sk_buff *skb)
 	}
 #endif
 
-	if (IS_ERR(ret))
-		return send_drop_notify(skb, SECLABEL, 0, 0, 0, ret, TC_ACT_SHOT);
-	else
+	if (IS_ERR(ret)) { // dropped
+		return send_drop_notify(skb, SECLABEL, 0, 0, 0, ret, TC_ACT_SHOT,
+		                        DIRECTION_EGRESS);
+	} else { //forwarded
+	    update_data_metrics(DIRECTION_EGRESS, 0);
```

Comment:
This is incorrect, there are various forwarding paths where we tail call and never reach this pointed even when forwarding. Look at the invocations of `DBG_CAPTURE_DELIVERY` to see all places where we forward a packet.

Reply:
Sounds good, thanks for the pointer.

Reply:
@tgraf :Also  `DBG_CAPTURE_DELIVERY` is applicable for `ingress` as well as `egress`?
So the forwarded count code added in `handle_policy` is incorrect for ingress?

Reply:
@joestringer @tgraf : What will the direction be for`cilium_dbg_capture(skb, DBG_CAPTURE_DELIVERY, skb->ifindex)`  in  `icmp6_send_reply`. and `arp_respond`. I think it should be EGRESS.
Similarly, going through `bpf_overlay.c`, `cilium_dbg_capture(skb, DBG_CAPTURE_DELIVERY, skb->ifindex);` in this case both should be egress, while in `bpf_lb.c` its ingress correct?

Reply:
Spoke to @rlenglet. We can use `send_trace_notify` instead to specifically capture forwarded packets.

Reply:
done